### PR TITLE
Closes #873 and revises #874

### LINKED ISSUE #873: Wrong entity offsets in the tmvar_v3 datasets
## Describe the bug
Some entities in the tmvar_v3 dataset have wrong entity offsets

## Steps to reproduce the bug

<details>
  <summary>Python Code</summary>

```python
from datasets import load_dataset

dataset_name= "tmvar_v3"
dataset = load_dataset(f"bigbio/{dataset_name}", name=f"{dataset_name}_bigbio_kb")

doc_id = "21904390"

def check_offsets(doc_id):
    text = dataset[split].filter(lambda x: x["document_id"] == doc_id)[0]["passages"][0]["text"][0] \
        + " " + dataset[split].filter(lambda x: x["document_id"] == doc_id)[0]["passages"][1]["text"][0]

    sentences = text.split(". ")

    sentence_indexes = [m.start() + 2 for m in re.finditer("\. ", text)]  # because the suffix is ". "
    sentence_indexes = [0] + sentence_indexes

    doc_entities = dataset[split].filter(lambda x: x["document_id"] == doc_id)[:]["entities"][0]

    print(sentence_indexes)
    print(len(sentences))
    print(text)
    print(doc_entities)

    sentence_index = 0
    entity_index = 0
    current_offset = 0
    next_sentence_offset = 0
    next_entity_offset = text[current_offset:].find(doc_entities[entity_index]["text"][0])
    while True:
        if sentence_index >= len(sentence_indexes) and entity_index >= len(doc_entities):
            break
        if next_sentence_offset <= next_entity_offset:
            sentence_end = sentence_indexes[sentence_index] + len(sentences[sentence_index]) + 2
            print(f"Sentence {sentence_index} @ offsets {sentence_indexes[sentence_index]} to {sentence_end}")
            print(sentences[sentence_index] + ". ")
            sentence_index += 1
            if sentence_index >= len(sentence_indexes):
                next_sentence_offset = len(text)
            else:
                next_sentence_offset = sentence_indexes[sentence_index]
            # print(f"DEBUG next_sentence_offset: {next_sentence_offset}")
        else:  # next_entity_offset < next_sentence_offset
            entity = doc_entities[entity_index]
            entity_name = entity["text"][0]
            given_offset_start = entity["offsets"][0][0]
            given_offset_end = entity["offsets"][0][1]
            print(f"  {entity_name} @ offsets (real) {next_entity_offset} to {next_entity_offset + len(entity_name)}")
            print(f"  {text[given_offset_start:given_offset_end]} @ offset (given) {given_offset_start} to {given_offset_end}")
            current_offset = next_entity_offset + len(entity_name)
            entity_index += 1
            if entity_index >= len(doc_entities):
                next_entity_offset = len(text)
            else:
                next_entity_offset = current_offset + text[current_offset:].find(doc_entities[entity_index]["text"][0])
            # print(f"DEBUG next_entity_offset: {next_entity_offset}")

check_offsets(doc_id)
```
</details>

## Expected results
In Pubmed ID 21904390, expected offsets for these seven entities "entity name, (offset_start, offset_end)" are as follows: 

```python
PAX6 (342, 346)
PAX6 (751, 755)
PAX6 (1153, 1157)
PAX6 (1483, 1487)
PAX6 (1627, 1631)
DKFZ p686k1684 (1640, 1654)
PAX6 (2037, 2041).
```

## Actual results
Offsets in the tmvar_v3 dataset for the seven entities are as follows:

```python
PAX6 (343, 347)
PAX6 (753, 757)
PAX6 (1156, 1160)
PAX6 (1487, 1491)
PAX6 (1631, 1635)
DKFZ p686k1684 (1645, 1659)
PAX6 (2043, 2047)
```

All the remaining entity offsets in the tmvar_v3 dataset seem to be correct.

### PR DESCRIPTION
**Pull request by WangXII**
Closes https://github.com/bigscience-workshop/biomedical/issues/873 and revises https://github.com/bigscience-workshop/biomedical/pull/874 - Wrong entity offsets in the tmvar_v3 datasets

Wrong offsets in PMID 21904390 are already present in the source file https://ftp.ncbi.nlm.nih.gov/pub/lu/tmVar3/tmVar3Corpus.txt

Solution: Manually corrected the wrong offsets in PMID 21904390 as the wrong offsets do not seem to follow any pattern.

Compared to https://github.com/bigscience-workshop/biomedical/pull/874, this pull request reverts the offsets in the standard 'source' dataset back to original (but wrong) offsets provided by the original dataset and adds a new 'source_fixed' dataset with corrected offsets

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio_hub <dataset_name> [--data_dir /path/to/local/data] --test_local`.

### CODE REVIEW

File: bigbio/hub/hub_repos/tmvar_v3/tmvar_v3.py [hunk 1]
```diff
@@ -269,13 +304,16 @@ def pubtator_to_bigbio_kb(self, filepath):
                         "id": next(uid),
                         "offsets": [[mention.start, mention.end]],
                         "text": [mention.text],
-                        "type": [mention.type],
```
Comment (leonweber) [inline bigbio/hub/hub_repos/tmvar_v3/tmvar_v3.py:272]: Same here: Is the change from sequence to single value intended here?

**Review by leonweber — CHANGES_REQUESTED** [review]
Imho this can be merged after the two comments have been resolved. Please also update to the lastest version of main. At the moment there is a conflict that prevents merging.

**Comment by WangXII** [comment]
Thanks for the review @leonweber. Looking into the merge conflict, the branch is already up-to-date with main. It seems more like a simple change that the automatic merging strategies fails to recognize. The branch version adds an if-clause to just show the fixed annotations in the source version if the user explicitly requests it ("_source_fixed" vs "_source")

```
<<<<<<< tmvar_v3_fix
                if "_fixed" in self.config.name:
                    document["entities"] = self._correct_wrong_offsets(
                        document["entities"], doc.pmid
                    )
=======
                document["entities"] = self._correct_wrong_offsets(
                    document["entities"], doc.pmid
                )
>>>>>>> main
```

File: bigbio/hub/hub_repos/tmvar_v3/tmvar_v3.py [hunk 2]
```diff
@@ -227,7 +258,7 @@ def pubtator_to_source(self, filepath):
                     {
                         "offsets": [[mention.start, mention.end]],
                         "text": [mention.text],
-                        "semantic_type_id": [mention.type],
```
[the remark this answers is not in the document]
Comment (WangXII) [reply bigbio/hub/hub_repos/tmvar_v3/tmvar_v3.py:230]: The goal is to bring the datatype of semantic_type_id into line with all other datasets where only a single value is used to hold the type (see https://github.com/bigscience-workshop/biomedical/blob/62fb81c8b24f692fbae675d11c545dd31dbc3eb5/bigbio/hub/hub_repos/bioinfer/bioinfer.py#L124 and https://github.com/bigscience-workshop/biomedical/blob/62fb81c8b24f692fbae675d11c545dd31dbc3eb5/bigbio/hub/hub_repos/nlm_gene/nlm_gene.py#L123). I've also checked that the dataset does not contain annotations with multiple semantic_type_id to be the reason for the sequence. However, if there was a rationale to let some semantic_type_id be sequences and some not then we can easily revert this change.

**Review by leonweber — APPROVED** [review]

**Comment by leonweber** [comment]
@mariosaenger Can you review this please?

**Review by mariosaenger — APPROVED** [review]

**Pull request merged**